packages/provider/src/tasks/client/clientTasks.ts

```diff
@@ -249,6 +249,48 @@ export class ClientTaskManager {
 		await this.providerDB.storeUserBlockRuleRecords(rules);
 	}
 
+	isSubdomainOrExactMatch(referrer: string, clientDomain: string): boolean {
+		if (!referrer || !clientDomain) return false;
+		try {
+			const cleanReferrer = referrer.toLowerCase().trim().replace(/\/+$/, "");
+			const cleanAllowedInput = clientDomain
+				.toLowerCase()
+				.trim()
+				.replace(/\/+$/, "");
+
+			// Extract domain from URL if necessary
+			const getDomain = (url: string) => {
```

Comment:
Please replace or use https://github.com/prosopo/captcha/blob/ed6d1a72c1f839f2d94c45d74df3633c145f172e/packages/util/src/url.ts#L21-L40 instead. Could just return the new URL thing in a new function and keep `validateDomain` separate. The main requirement is having a single domain parser that we use across portal / lambdas / provider. This sort of thing should be in util.